onnxruntime/core/providers/cpu/math/matmul_integer.cc

```diff
@@ -52,35 +51,61 @@ Status MatMulInteger<uint8_t, uint8_t, int32_t>::Compute(OpKernelContext* ctx) c
   ORT_RETURN_IF_ERROR(helper.Compute(a->Shape(), b->Shape()));
   Tensor* y = ctx->Output(0, helper.OutputShape());
 
-  // validate zero points
-  int32_t a_offset = 0;
-  int32_t b_offset = 0;
-  if (has_a_zero_point_) {
-    auto a_zero_point = ctx->Input<Tensor>(2);
-    ORT_ENFORCE(a_zero_point->Shape().NumDimensions() == 0 ||
-        (a_zero_point->Shape().NumDimensions() == 1 && a_zero_point->Shape().GetDims().size() == 1),
-        "Currently only scalar zero_point is supported. TODO: add per channel zero point support.");
-    a_offset = static_cast<int32_t>(*a_zero_point->template Data<uint8_t>());
-  }
-  if (has_b_zero_point_) {
-    auto b_zero_point = ctx->Input<Tensor>(3);
-    ORT_ENFORCE(b_zero_point->Shape().NumDimensions() == 0 ||
-        (b_zero_point->Shape().NumDimensions() == 1 && b_zero_point->Shape().GetDims().size() == 1),
-        "Currently only scalar zero_point is supported. TODO: add per channel zero point support.");
-    b_offset = static_cast<int32_t>(*b_zero_point->template Data<uint8_t>());
-  }
+  ORT_RETURN_IF_NOT(y->DataType() == DataTypeImpl::GetType<std::int32_t>());
```

Comment:
>ORT_RETURN_IF_NOT [](start = 2, length = 17)

Is this not handled by the onnx constraint checks?

Reply:
It should. The code seems not necessary.

---
In reply to: [312417990](https://github.com/microsoft/onnxruntime/pull/881#discussion_r312417990) [](ancestors = 312417990)